packages/geo/src/lib/import-export/import-export/import-export.component.ts

```diff
@@ -552,36 +562,88 @@ export class ImportExportComponent implements OnDestroy, OnInit {
+              if (data.separator) {
+                if (previousFeature) {
+                  if (currentFeature.get('type') !== previousFeature.get('type')) {
+                    const emptyRowFeature = previousFeature.clone();
+                    const previousFeatureKeys: Array<string> = previousFeature.getKeys();
+                    previousFeatureKeys.forEach(key => {
+                      emptyRowFeature.unset(key, true);
+                    });
+                    featuresCSV.push(emptyRowFeature);
+                  }
+                }
+              }
+              featuresCSV.push(currentFeature);
+              previousFeature = currentFeature;
+            });
+          });
+        }
+      }
+
       if (geomTypes.length === 0) {
         this.loading$.next(false);
         const title = translate.instant('igo.geo.export.nothing.title');
         const message = translate.instant('igo.geo.export.nothing.text');
         this.messageService.error(message, title, { timeOut: 20000 });
 
       } else {
-        geomTypes.map(geomType =>
-          this.exportService.export(geomType.features, data.format, filename + geomType.geometryType, data.encoding, this.map.projection)
-          .subscribe(
-            () => {},
-            (error: Error) => this.onFileExportError(error),
-            () => {
-              this.onFileExportSuccess();
-
-              geomType.features.forEach(feature => {
-                const radius: number = feature.get('rad');
-
-                if (radius) {
-                  const point = new olPoint([feature.get('longitude'), feature.get('latitude')]);
-                  point.transform('EPSG:4326', feature.get('_projection'));
-                  feature.setGeometry(point);
-                }
-              });
-
-              this.loading$.next(false);
+        if (!(data.format === ExportFormat.CSVsemicolon || data.format === ExportFormat.CSVcomma) || !data.combineLayers) {
+          geomTypes.map(geomType =>
+            this.exportService.export(geomType.features, data.format, filename + geomType.geometryType, data.encoding, this.map.projection)
+            .subscribe(
+              () => {},
+              (error: Error) => this.onFileExportError(error),
+              () => {
+                this.onFileExportSuccess();
+
+                geomType.features.forEach(feature => {
+                  const radius: number = feature.get('rad');
+
+                  if (radius) {
+                    const point = new olPoint([feature.get('longitude'), feature.get('latitude')]);
+                    point.transform('EPSG:4326', feature.get('_projection'));
+                    feature.setGeometry(point);
+                  }
+                });
+
+                this.loading$.next(false);
             }
-        ));
+          ));
+        }
       }
-    });
+    };
+    if ((data.format === ExportFormat.CSVsemicolon || data.format === ExportFormat.CSVcomma) && data.combineLayers) {
```

Comment:
Should be put into else condition (line 561) ?

Reply:
Done